com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeVolumeSceneData.cs

```diff
@@ -76,6 +77,31 @@ internal class BakingSet
         internal Dictionary<string, ProbeVolumeBakingProcessSettings> sceneBakingSettings;
         internal List<BakingSet> bakingSets;
 
+        [SerializeField] internal List<string> bakingStates;
+        [SerializeField] private int m_BakingState = -1;
+
+        /// <summary>Delegate for baking state change.</summary>
+        /// <param name="newState">The new baking state.</param>
+        public delegate void BakingStateChangedDelegate(int newState);
+        /// <summary>Delegate called when the baking state is changed. </summary>
+        public static BakingStateChangedDelegate onBakingStateChanged;
+
+        /// <summary>The currently selected baking state.</summary>
+        public static int bakingState
+        {
+            get => ProbeReferenceVolume.instance.sceneData.m_BakingState;
+            set
+            {
+                var sceneData = ProbeReferenceVolume.instance.sceneData;
+                if (value == sceneData.m_BakingState)
+                    return;
+                sceneData.m_BakingState = value;
+                foreach (var data in Object.FindObjectsOfType<ProbeVolumePerSceneData>())
```

Comment:
This is really bad to be called at runtime Object.FindObjectsOfType<ProbeVolumePerSceneData>(). 

We should have the per scene data register itself somewhere upon loading and then go through the registered ones.